client/profile-wizard/index.js

```diff
@@ -243,28 +239,12 @@ class ProfileWizard extends Component {
 			} ),
 		];
 
-		let redirectUrl = null;
-		if ( shouldConnectJetpack ) {
-			promises.push(
-				getJetpackConnectUrl( {
-					redirect_url: getAdminLink( 'admin.php?page=wc-admin' ),
-				} ).then( ( jetpackConnectUrl ) => {
-					const error = getPluginsError( 'getJetpackConnectUrl' );
-					if ( error ) {
-						createNoticesFromResponse( error );
-						return;
-					}
-					redirectUrl = jetpackConnectUrl;
-				} )
-			);
-		}
-
 		Promise.all( promises ).then( () => {
-			if ( redirectUrl ) {
-				window.location = redirectUrl;
-				return;
+			if ( shouldConnectJetpack ) {
```

Comment:
Just want to note that this introduces blocking requests so it will take a performance hit, albeit marginal, by starting the Jetpack install/activation/connection after other promises have resolved.

We may want to consider a loading state in a follow-up if we go this route since on slower hosts this will result in long white screen.

Reply:
I'm trying to figure out how this is more blocking than before. The only real difference is that before we ran 2 requests in parallel but the user still had to wait till both had completely resolved before an actual redirect happened? The code that shows the loading state still happens first, and no sooner than before?

Reply:
>The only real difference is that before we ran 2 requests in parallel

This is the main reason.  Running those in parallel seemed to cut down the time pretty significantly on slower hosts where we had complaints around plugin installation and Jetpack connection.

If each request took 4 seconds, running in parallel means completion in 4 seconds, while running synchronously would be 8 seconds.

But the improvements in logic and testability are great here and may be a good tradeoff for the performance loss, I just want to make sure we discuss and acknowledge it before merging.

Reply:
@joshuatf gotcha, that makes sense. Yeah you're right its a trade off. I do wonder if there are other ways we could get performance back in future via further refactoring. I did the minimal work here to get this particular code working without `__experimentalResolveSelect` but I do think there may be ways to optimise.